string name
string role_filler